src/base/os.lua

```diff
@@ -387,10 +391,17 @@
 				while os.matchnext(m) do
 				if not (slashpos and os.matchisfile(m)) then
 					local matchpath = path.join(before, matchpath, os.matchname(m))
+					
 					if after then
-						results = table.join(results, os.match(path.join(matchpath, after)))
+						results = table.join(results, os.match(path.join(matchpath, after), matchType))
 					else
-						table.insert(results, matchpath)
+						if matchType == "file" and os.matchisfile(m) then
+							table.insert(results, matchpath)
+						elseif matchType == "folder" and not os.matchisfile(m) then
+							table.insert(results, matchpath)
+						else -- keep previous behaviour
```

Comment:
I just realised it's this here, it needs to check that `matchType == nil` or something, otherwise all folders when matching files are picked up, and all files when matching folders are picked up.

Reply:
In addition the comment is not meaningful outside of the pull request.